### PR TITLE
data-base-migration

### PR DESCRIPTION
**Pull request by Nathanael-lscx**
### Description
+ Rebuilding structure : adapt database and documentation to eachother
+ Adding async and migration
+ First requests working

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the documentatioe, if necessary

### CODE REVIEW

**Review by Victor-Angot — APPROVED** [review]
Don't forget to add documentation especially on the alambic part

File: app/main.py [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+
 app = FastAPI()
 
+# Décommenter cette partie en dev pour créer la base de donnée la première fois
+# @app.on_event("startup")
+# async def startup():
+#     # create db tables
+#     async with engine.begin() as conn:
+#         # await conn.run_sync(Base.metadata.drop_all)
+#         await conn.run_sync(Base.metadata.create_all)
```
Comment (Jhobahtes) [inline app/main.py:18]: Could you modify this part?

**Review by Jhobahtes — CHANGES_REQUESTED** [review]

**Comment by Jhobahtes** [comment]
TODO:
 * [ ] Check dependencies (greenlet probably missing)

File: app/endpoints/groups.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+@router.get(
+    "/groups/",
+    response_model=list[schemas_core.CoreGroupSimple],
+    status_code=200,
+    tags=[Tags.groups],
+)
+async def get_groups(db: AsyncSession = Depends(get_db)):
+    """Return all groups from database as a list of dictionaries"""
```
Comment (Jhobahtes) [inline app/endpoints/groups.py:21]: The returned object is more a list of `schemas_core.CoreGroupSimple` than a dictionary.

```suggestion
    """Return all groups"""
```

File: tests/test_users.py [hunk 3]
```diff
@@ -28,22 +31,21 @@ def override_get_db():
 
 app.dependency_overrides[get_db] = override_get_db
 
-client = TestClient(app)
+client = TestClient(app)  # Run the tests
```
Comment (Jhobahtes) [inline tests/test_users.py:34]: This line does not really run the tests. The idea is more to create a test client that is able to execute requests

**Comment by Jake-Ransom** [comment]
> TODO:

**Comment by Jake-Ransom** [comment]
la glissade

**Review by Jhobahtes — APPROVED** [review]

**Review by Victor-Angot — APPROVED** [review]

**Pull request merged**